src/Crawler.php

```diff
@@ -178,68 +181,52 @@ protected function getCrawlRequests(): Generator
+            yield new Request('GET', $crawlUrl->url);
             $i++;
         }
     }
 
-    protected function addAllLinksToCrawlQueue(string $html, Url $foundOnUrl)
+    protected function addAllLinksToCrawlQueue(string $html, UriInterface $foundOnUrl)
     {
-        $allLinks = $this->extractAllLinks($html);
+        $allLinks = $this->extractAllLinks($html, $foundOnUrl);
 
         collect($allLinks)
-            ->filter(function (Url $url) {
-                return $url->hasCrawlableScheme();
+            ->filter(function (UriInterface $url) {
+                return in_array($url->getScheme(), ['http', 'https']);
             })
-            ->map(function (Url $url) use ($foundOnUrl) {
-                return $this->normalizeUrl($url, $foundOnUrl);
+            ->map(function (UriInterface $url) {
+                return $this->normalizeUrl($url);
             })
-            ->filter(function (Url $url) {
+            ->filter(function (UriInterface $url) {
                 return $this->crawlProfile->shouldCrawl($url);
             })
             ->reject(function ($url) {
                 return $this->crawlQueue->has($url);
             })
-            ->each(function (Url $url) use ($foundOnUrl) {
+            ->each(function (UriInterface $url) use ($foundOnUrl) {
                 $this->crawlQueue->add(
                     CrawlUrl::create($url, $foundOnUrl)
                 );
             });
     }
 
-    protected function extractAllLinks(string $html): Collection
+    protected function extractAllLinks(string $html, UriInterface $foundOnUrl): Collection
     {
-        $domCrawler = new DomCrawler($html);
+        $domCrawler = new DomCrawler($html, $foundOnUrl);
 
-        return collect($domCrawler->filterXpath('//a')->extract(['href']))
-            ->map(function ($url) {
-                return Url::create($url);
+        return collect($domCrawler->filterXpath('//a')->links())
+            ->map(function (Link $link) {
+                return new Uri($link->getUri());
             });
     }
 
     /**
-     * @param \Spatie\Crawler\Url $url
-     * @param \Spatie\Crawler\Url $foundOnUrl
+     * @param UriInterface $url
      *
-     * @return \Spatie\Crawler\Url
+     * @return UriInterface
      */
-    protected function normalizeUrl(Url $url, Url $foundOnUrl): Url
+    protected function normalizeUrl(UriInterface $url): UriInterface
     {
-        if ($url->isRelativeToPath()) {
-            $directory = $foundOnUrl->directory();
-            $url->setPath($directory.$url->path());
-        }
-
-        if ($url->isRelative()) {
-            $url->setScheme($this->baseUrl->scheme)
-                ->setHost($this->baseUrl->host)
-                ->setPort($this->baseUrl->port);
-        }
-
-        if ($url->isProtocolIndependent()) {
-            $url->setScheme($this->baseUrl->scheme);
-        }
-
-        return $url->removeFragment();
+        return $url->withFragment('');
```

Comment:
Return new Uri instance without the fragment. Rest is done by Symfony/UriInterface already